src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -328,6 +328,11 @@ BlockIO InterpreterDropQuery::executeToDatabaseImpl(const ASTDropQuery & query,
                 query_for_table.database = database_name;
                 query_for_table.no_delay = query.no_delay;
 
+                for (auto iterator = database->getTablesIterator(getContext()); iterator->isValid(); iterator->next())
```

Comment:
Probably we need to flush even if database->shouldBeEmptyOnDetach() is false.

Reply:
OK, let's flush globally, even though those databases (dictionary, psql, mysql) cannot have Buffer table